// @novu/node/build/module/lib/subscribers/subscribers.js
export class Subscribers {
    constructor(http) {
        this.http = http;
    }
    async list(page) {
        return await this.http.get(`/subscribers`, {
            params: {
                page,
            },
        });
    }
    async get(subscriberId) {
        return await this.http.get(`/subscribers/${subscriberId}`);
    }
    async identify(subscriberId, data) {
        return await this.http.post(`/subscribers`, {
            subscriberId,
            ...data,
        });
    }
    async update(subscriberId, data) {
        return await this.http.put(`/${subscriberId}/subscribers`, {
            ...data,
        });
    }
    async getPreference(subscriberId) {
        return await this.http.get(`/subscribers/${subscriberId}/preferences`);
    }
    async updatePreference(subscriberId, templateId, data) {
        return await this.http.patch(`/subscribers/${subscriberId}/preferences/${templateId}`, {
            ...data,
        });
    }
    async setCredentials(subscriberId, providerId, credentials) {
        return await this.http.put(`/subscribers/${subscriberId}/credentials`, {
            providerId,
            credentials: {
                ...credentials,
            },
        });
    }
    async delete(subscriberId) {
        return await this.http.delete(`/subscribers/${subscriberId}`);
    }
}
